sslr-core/src/test/java/org/sonar/sslr/internal/vm/PatternExpressionTest.java

```diff
@@ -85,4 +85,12 @@ public void should_catch_StackOverflowError() {
     expression.execute(machine);
   }
 
+  @Test
+  public void does_not_keep_reference() {
+    when(machine.length()).thenReturn(1);
+    when(machine.charAt(0)).thenReturn('z');
+    expression.execute(machine);
+    assertThat(expression.getMatcher().replaceAll("")).isEqualTo("");
+  }
+
```

Comment:
Possible other test:
```
    Machine machine = new Machine("abc", new Instruction[]{});
    WeakReference<Machine> reference = new WeakReference<>(machine);
    expression.execute(machine);
    machine = null;
    Runtime.getRuntime().gc();
    assertThat(reference.get()).isNull();
```
However, I'm not sure it would work reliably because of the call to `gc()`...

Reply:
> However, I'm not sure it would work reliably because of the call to gc()...

JVM can ignore `gc()` call.

Reply:
> JVM can ignore `gc()` call.

Hence my comment.
However, we should be able to more or less control the JVM which runs our unit tests.